presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,375 @@
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT, FieldType.LONG);
+        }
+        else if (columnType == INTEGER) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT);
+        }
+        else if (columnType == SMALLINT) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT);
+        }
+        else if (columnType == TINYINT) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+        }
+        else if (columnType == BOOLEAN) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE, FieldType.SHORT, FieldType.INT, FieldType.LONG);
+        }
+        else if (columnType == DOUBLE) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.DOUBLE, FieldType.FLOAT);
+        }
+        else if (isVarcharType(columnType)) {
+            checkFieldTypeOneOf(fieldType, columnName, columnType, FieldType.BYTE);
+        }
+    }
+
+    private void checkFieldTypeOneOf(FieldType declaredFieldType, String columnName, Type columnType, FieldType... allowedFieldTypes)
+    {
+        checkArgument(Arrays.asList(allowedFieldTypes).contains(declaredFieldType),
+                format("Wrong dataformat '%s' specified for column '%s'; %s type implies use of %s",
+                        declaredFieldType.name(),
+                        columnName,
+                        columnType,
+                        Joiner.on("/").join(allowedFieldTypes)));
+    }
+
+    private static boolean isSupportedType(Type type)
+    {
+        return isVarcharType(type) || SUPPORTED_PRIMITIVE_TYPES.contains(type);
+    }
+
+    @Override
+    protected void appendNullValue()
+    {
+        for (int i = 0; i < columnMappings.get(currentColumnIndex).getLength(); i++) {
+            buffer.put((byte) 0);
+        }
+    }
+
+    @Override
+    protected void appendLong(long value)
+    {
+        buffer.putLong(value);
+    }
+
+    @Override
+    protected void appendInt(int value)
+    {
+        buffer.putInt(value);
+    }
+
+    @Override
+    protected void appendShort(short value)
+    {
+        buffer.putShort(value);
+    }
+
+    @Override
+    protected void appendByte(byte value)
+    {
+        buffer.put(value);
+    }
+
+    @Override
+    protected void appendDouble(double value)
+    {
+        buffer.putDouble(value);
+    }
+
+    @Override
+    protected void appendFloat(float value)
+    {
+        buffer.putFloat(value);
+    }
+
+    @Override
+    protected void appendBoolean(boolean value)
+    {
+        buffer.put((byte) (value ? 1 : 0));
+    }
+
+    @Override
+    protected void appendString(String value)
+    {
+        byte[] valueBytes = value.getBytes(StandardCharsets.UTF_8);
+        checkArgument(valueBytes.length == columnMappings.get(currentColumnIndex).getLength(), format(
```

Comment:
This is a strong constraint. Maybe allowing for shorter strings with some user-defined padding character would give better ux?
@findepi @aalbu WDYT? (can be relaxed as a followup)

Reply:
I didn't want to change the specifications for the raw format in this pr. I agree with you that padding shorter strings would be a good idea. The specifications for the raw format need to be rethought and clearly defined in the docs, figured it would be best to do that in a follow-up pr.

Reply:
I agree with @charlesjmorgan.  Without being explicit in the specification, it's probably best to require the users to pad in the query.